contractcourt/channel_arbitrator.go

```diff
@@ -455,31 +456,22 @@ func (c *ChannelArbitrator) relaunchResolvers() error {
 	// to prevent a db migration. We use all available htlc sets here in
 	// order to ensure we have complete coverage.
 	htlcMap := make(map[wire.OutPoint]*channeldb.HTLC)
-	for _, htlcs := range c.activeHTLCs {
-		for _, htlc := range htlcs.incomingHTLCs {
-			htlc := htlc
-			outpoint := wire.OutPoint{
-				Hash:  commitHash,
-				Index: uint32(htlc.OutputIndex),
-			}
-			htlcMap[outpoint] = &htlc
-		}
-
-		for _, htlc := range htlcs.outgoingHTLCs {
-			htlc := htlc
-			outpoint := wire.OutPoint{
-				Hash:  commitHash,
-				Index: uint32(htlc.OutputIndex),
-			}
-			htlcMap[outpoint] = &htlc
+	for _, htlc := range confirmedHTLCs {
+		htlc := htlc
```

Comment:
I would expect this not to be possible, because the loop variable was already defined in this scope. But apparently it works. Was already like that I see in diff.

Reply:
What wouldn't be possible? Shadowing the loop variable? It's another way to solve the issue of binding to the loop variable when it's a pointer/slice.

Reply:
I was just surprised that there are two scopes inside a for loop, that's all.